File: packages/health_connector_hc_android/android/src/main/kotlin/com/phamtunglam/health_connector_hc_android/mappers/metadata_mappers/RecordingMethodMapper.kt
package com.phamtunglam.health_connector_hc_android.mappers.metadata_mappers

import androidx.health.connect.client.records.metadata.Metadata
import com.phamtunglam.health_connector_hc_android.pigeon.RecordingMethodDto

/**
 * Converts a Health Connect recording method constant to a [RecordingMethodDto].
 */
internal fun Int.toRecordingMethodDto(): RecordingMethodDto = when (this) {
    Metadata.RECORDING_METHOD_MANUAL_ENTRY -> RecordingMethodDto.MANUAL_ENTRY
    Metadata.RECORDING_METHOD_AUTOMATICALLY_RECORDED -> RecordingMethodDto.AUTOMATICALLY_RECORDED
    Metadata.RECORDING_METHOD_ACTIVELY_RECORDED -> RecordingMethodDto.ACTIVELY_RECORDED
    else -> RecordingMethodDto.UNKNOWN
}

File: packages/health_connector_hc_android/lib/src/mappers/metadata_mappers/recording_method_mapper.dart
import 'package:health_connector_core/health_connector_core.dart'
    show RecordingMethod, sinceV1_0_0;
import 'package:health_connector_hc_android/src/pigeon/health_connector_hc_android_api.g.dart'
    show RecordingMethodDto;
import 'package:meta/meta.dart' show internal;

/// Converts [RecordingMethod] to [RecordingMethodDto].
@sinceV1_0_0
@internal
extension RecordingMethodDtoMapper on RecordingMethod {
  RecordingMethodDto toDto() {
    switch (this) {
      case RecordingMethod.unknown:
        return RecordingMethodDto.unknown;
      case RecordingMethod.manualEntry:
        return RecordingMethodDto.manualEntry;
      case RecordingMethod.automaticallyRecorded:
        return RecordingMethodDto.automaticallyRecorded;
      case RecordingMethod.activelyRecorded:
        return RecordingMethodDto.activelyRecorded;
    }
  }
}

/// Converts [RecordingMethodDto] to [RecordingMethod].
@sinceV1_0_0
@internal
extension RecordingMethodDtoToDomain on RecordingMethodDto {
  RecordingMethod toDomain() {
    switch (this) {
      case RecordingMethodDto.unknown:
        return RecordingMethod.unknown;
      case RecordingMethodDto.manualEntry:
        return RecordingMethod.manualEntry;
      case RecordingMethodDto.automaticallyRecorded:
        return RecordingMethod.automaticallyRecorded;
      case RecordingMethodDto.activelyRecorded:
        return RecordingMethod.activelyRecorded;
    }
  }
}
